Modules/_datetimemodule.c

```diff
@@ -6924,18 +6935,12 @@ _datetime_exec(PyObject *module)
     }
 
     /* At last, set up and add the encapsulated C API */
-    PyDateTime_CAPI *capi = get_datetime_capi();
-    if (capi == NULL) {
-        goto error;
-    }
-    PyObject *capsule = PyCapsule_New(capi, PyDateTime_CAPSULE_NAME,
-                                      datetime_destructor);
+    set_datetime_capi(st);
+    PyObject *capsule = PyCapsule_New(&st->capi, PyDateTime_CAPSULE_NAME, NULL);
```

Comment:
Why do you pass a `NULL` as a capsule destructor?

Reply:
In this case, the cupsule just views a memory allocated and freed by the moduleobject.
https://github.com/python/cpython/blob/bfc57d43d8766120ba0c8f3f6d7b2ac681a81d8a/Objects/moduleobject.c#L739-L740